CHANGELOG.md

```diff
@@ -4,6 +4,207 @@ All notable changes to Router will be documented in this file.
 
 This project adheres to [Semantic Versioning v2.0.0](https://semver.org/spec/v2.0.0.html).
 
+# [1.48.0] - 2024-05-29
+
+## 🚀 Features
+
+### Preview demand control ([PR #5317](https://github.com/apollographql/router/pull/5317))
+
+> Note that Demand Control is in preview. Configuration and performance may change in future releases.
+Demand control is a feature that allows you to control the cost of operations in the router, potentially rejecting requests that are too expensive that could bring down the Router or subgraph services.
+
+```yaml
+# Demand control enabled, but in measure mode.
+preview_demand_control:
+  enabled: true
+  # `measure` or `enforce` mode. Measure mode will analyze cost of operations but not reject them.
+  mode: measure
+
+  strategy:
+    # Static estimated strategy has a fixed cost for elements and when set to enforce will reject
+    # requests that are estimated as too high before any execution takes place.
+    static_estimated:
+      # The assumed returned list size for operations. This should be set to the maximum number of items in graphql list
+      list_size: 10
+      # The maximum cost of a single operation.
+      max: 1000
+```
+
+Telemetry is emitted for demand control, including the estimated cost of operations and whether they were rejected or not.
+Full details will be included in the documentation for demand control which will be finalized before the next release.
+
+By [@bryncooke](https://github.com/bryncooke) in https://github.com/apollographql/router/pull/5317
+
+### Add support for studio trace id in selectors and document context entry ([Issue #3803](https://github.com/apollographql/router/issues/3803)), ([Issue #5172](https://github.com/apollographql/router/issues/5172))
+
+Add support for a new trace ID selector kind, the `apollo` trace ID, which represents the trace ID on [Apollo GraphOS Studio](https://studio.apollographql.com/). 
+
+An example configuration using `trace_id: apollo`:
+
+```yaml
+telemetry:
+  instrumentation:
+    spans:
+      router:
+        "studio.trace.id":
+            trace_id:: apollo
+```
+
+Also, router documentation has been updated with constants that are available in Rhai.
```

Comment:
@bnjjj This is really valuable!  But doesn't seem related to **this** changelog entry at all which makes it quite confusing to me. Can we just add a separate changelog entry that puts a "Documentation" update in?  It looks like it refers to:

https://github.com/apollographql/router/pull/5189/files#diff-5888944c765e2b49417fb75dd8c0e518bf4cb68fc264e6963f5be2da03503402R340-R352

Reply:
After I submit this review, you'll see that I've made the suggestion below to fix this, therefore, I'll suggest removing this part, too (and update this comment with a link, afterwards).

Reply:
Yes you're right, @Geal you can apply these changes